extensions/avro-extensions/src/main/java/io/druid/data/input/schemarepo/SubjectAndIdConverter.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Druid - a distributed column store.
+ * Copyright 2012 - 2015 Metamarkets Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.druid.data.input.schemarepo;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.metamx.common.Pair;
+import org.schemarepo.api.converter.Converter;
+
+import java.nio.ByteBuffer;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type", defaultImpl = Avro1124SubjectAndIdConverter.class)
+@JsonSubTypes(value = {
+    @JsonSubTypes.Type(name = "avro-1124", value = Avro1124SubjectAndIdConverter.class)
+})
+public interface SubjectAndIdConverter<SUBJECT, ID>
+{
+
+  Pair<SUBJECT, ID> toSubjectAndId(ByteBuffer payload);
+
+  void fromSubjectAndId(SUBJECT subject, ID id, ByteBuffer payload);
+
```

Comment:
[get/put]SubjectAndId(..) are probably more appropriate  names for these 2. Also can you add some javadocs on this interface e.g. in what situation would someone need to add another implementation?